src/main/java/io/ebeaninternal/server/query/SqlTreeNodeBean.java

```diff
@@ -135,6 +135,11 @@ private SqlTreeNodeBean(String prefix, BeanPropertyAssoc<?> beanProp, BeanDescri
 
   @Override
   public BeanProperty getSingleProperty() {
+    if (properties == null || properties.length == 0) {
+      // if we have no property ask first children (in a distinct select with join)
+      // if we have also no children, NPE happens anyway.
+      return children[0].getSingleProperty();
+    }
     return properties[0];
```

Comment:
if we want a single property, we use the first one we can find - while writing this comment, it could make sense to iterate over all children until we find one.